bindings/ergo-lib-wasm/examples/address-generation-demo/index.ts

```diff
@@ -0,0 +1,50 @@
+import {
+    ExtSecretKey,
+    DerivationPath,
+    NetworkAddress,
+    NetworkPrefix,
+    Mnemonic
+} from '../../pkg-nodejs/ergo_lib_wasm';
+
+const secretSeedFromMnemonic = (mnemonic: string): Uint8Array =>
+    Mnemonic.to_seed(mnemonic, "");
+
+const masterSecretFromSeed = (seed: Uint8Array) =>
+    ExtSecretKey.derive_master(seed);
+
+const deriveSecretKey = (rootSecret: ExtSecretKey, path: DerivationPath) =>
+    rootSecret.derive(path); 
+
+const nextPath = (rootSecret: ExtSecretKey, lastPath: DerivationPath) => 
+    rootSecret.derive(lastPath).path().next();
+
+const main = () => {
+    const mnemonic = "change me do not use me change me do not use me";
+
+    const seed = secretSeedFromMnemonic(mnemonic);
+    const rootSecret = masterSecretFromSeed(seed);
+
+    // This is using EIP-3 pathing: m/44'/429'/0'/0/0
+    // First param is 0' (account) and the last 0 (address index)
+    let changePath = DerivationPath.new(0, new Uint32Array([0]));
+    const changeSecretKey = deriveSecretKey(rootSecret, changePath);
+    const changePubKey = changeSecretKey.public_key();
+    const changeAddress = NetworkAddress.new(NetworkPrefix.Mainnet, changePubKey.to_address());
+
+    console.log(`Change address: ${changeAddress.to_base58()}`);
+
+    // This is currently required becuase the line:
+    // `const changeSecretKey = deriveSecretKey(rootSecret, changePath);`
+    // Takes ownership of the changePath pointer and frees it so it's null when we get to this point
```

Comment:
Maybe we need to use references a bit more in wasm bindings?

Reply:
Oh yes, I use the references almost always because of that. I try to catch them in review as well, but sometimes that get through.

Reply:
That's likely my bad, I'll be using these wasm bindings a heap so I'll fix them up as I come across them!